### PR TITLE
Task: Reassessment Credit Modifications #2195

### PR DESCRIPTION
**Pull request by JulianForeman**
- Allowing all credit fields to be modified in reassessments
- Added checks for empty values to avoid errors

### CODE REVIEW

**Review by tim738745 — CHANGES_REQUESTED** [review]

File: frontend/src/supplementary/components/CreditActivity.js [hunk 1]
```diff
@@ -488,7 +488,7 @@ const CreditActivity = (props) => {
                               &bull; &nbsp; &nbsp; {deduction.modelYear} Credits
                             </td>
                             <td className="text-right">
-                              {getAssociatedDeduction(deduction, deductions)
+                              {creditA && AgetAssociatedDeduction(deduction, deductions)
```
[the remark this answers is not in the document]
Comment (JulianForeman) [reply frontend/src/supplementary/components/CreditActivity.js:491]: Yes I did haha

**Comment by JulianForeman** [comment]
> Also, @JulianForeman, just wondering if you tested making changes to the newly unlocked credit fields in a reassessment and submitting the reassessment? And upon director issuance, is the supplier's credit balance updated? Thanks!

The process seems to work 100% through saving, submitting and issuing the reassessment. I'm not too sure how to read the credit balances so I can't say for certain or not.

File: frontend/src/supplementary/components/CreditActivity.js [hunk 2]
```diff
@@ -514,32 +514,32 @@ const CreditActivity = (props) => {
                                   )}
                                 </span>
                               )}
-                              {getAssociatedDeduction(deduction, deductions)
+                              {creditB && getAssociatedDeduction(deduction, deductions)
```
Comment (tim738745) [inline frontend/src/supplementary/components/CreditActivity.js:517]: Just one more here; should be: `getAssociatedDeduction(deduction, deductions).creditB && getAssociatedDeduction(deduction, deductions).creditB.eq(bigZero)`

**Review by tim738745 — CHANGES_REQUESTED** [review]

**Comment by tim738745** [comment]
> > Also, @JulianForeman, just wondering if you tested making changes to the newly unlocked credit fields in a reassessment and submitting the reassessment? And upon director issuance, is the supplier's credit balance updated? Thanks!
> 
> The process seems to work 100% through saving, submitting and issuing the reassessment. I'm not too sure how to read the credit balances so I can't say for certain or not.

Ok, thanks!

**Review by tim738745 — APPROVED** [review]

**Pull request merged**